Core/Core/Router/Router.swift

```diff
@@ -18,43 +18,40 @@
 
 import UIKit
 
-public struct RouteOptions: Equatable {
-    public let modal: Modal?
-    public let detail: Bool
-    public let embedInNav: Bool
-    public let addDoneButton: Bool
-
-    public struct Modal: Equatable {
-        public let presentationStyle: UIModalPresentationStyle?
-        public let inPresentation: Bool
-    }
-
-    public init(
-        modal: Modal? = nil,
-        detail: Bool = false,
+public enum RouteOptions: Equatable {
+    case push(addDoneButton: Bool = false)
```

Comment:
I don't know if push needs an addDoneButton. I'm pretty sure we only do that for modals, but I didn't look at all of the uses to see if it does show up.